File: Quorum3/Library/Standard/Plugins/javascript/Libraries/Compute/BitwiseOperations.js
function plugins_quorum_Libraries_Compute_BitwiseOperations_() {
    this.And$quorum_integer$quorum_integer = function (left, right) {
        return left & right;
    };
    
    this.Or$quorum_integer$quorum_integer = function (left, right) {
        return left | right;
    };
    
    this.ExclusiveOr$quorum_integer$quorum_integer = function (left, right) {
        return left ^ right;
    };
    
    this.Negate$quorum_integer = function (value) {
        return ~value;
    };
    
    this.ShiftLeft$quorum_integer$quorum_integer = function (value, amount) {
        return value << amount;
    };
    
    this.ShiftRight$quorum_integer$quorum_integer = function (value, amount) {
        return value >> amount;
    };
    
    this.ShiftRightPositive$quorum_integer$quorum_integer = function (value, amount) {
        return value >>> amount;
    };
    
}

File: Plugins/MathPlugin/src/plugins/quorum/Libraries/Compute/BitwiseOperations.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Compute;

/**
 *
 * @author astefik
 */
public class BitwiseOperations {
    public java.lang.Object me_ = null;
    
    public int And(int left, int right){
        return left & right; 
    }
    
    public int Or(int left, int right) {
        return left | right;
    }
    public int ExclusiveOr(int left, int right) {
        return left ^ right;
    }
    public int Negate(int value) {
        return ~value;
    }
    public int ShiftLeft(int value, int amount) {
        return value << amount;
    }
    public int ShiftRight(int value, int amount) {
        return value >> amount;
    }
    public int ShiftRightPositive(int value, int amount) {
        return value >>> amount;
    }
}
